artifactory/commands/replication/template.go

```diff
@@ -0,0 +1,223 @@
+		},
+		Msg:          "Select the template type",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       nil,
+		MapKey:       "",
+		Callback:     nil,
+	},
+	JobType: {
+		Options: []prompt.Suggest{
+			{Text: Pull, Description: "Pull replication job"},
+			{Text: Push, Description: "Push replication job"},
+		},
+		Msg:          "Select job type",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       JobType,
+		Callback:     jobTypeCallback,
+	},
+	RepoKey: {
+		Msg:          "Enter repo key",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       RepoKey,
+		Callback:     nil,
+	},
+	Username: {
+		Msg:          "Enter user name",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       Username,
+		Callback:     nil,
+	},
+	Password: {
+		Msg:          "Enter password",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       Password,
+		Callback:     nil,
+	},
+	URL: {
+		Msg:          "Enter URL",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       URL,
+		Callback:     nil,
+	},
+	CronExp: {
+		Msg:          "Enter cron expression",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       CronExp,
+		Callback:     nil,
+	},
+	EnableEventReplication: utils.BoolQuestionInfo,
+	Enabled:                utils.BoolQuestionInfo,
+	SyncDeletes:            utils.BoolQuestionInfo,
+	SyncProperties:         utils.BoolQuestionInfo,
+	SyncStatistics:         utils.BoolQuestionInfo,
+	PathPrefix: {
+		Msg:          "Enter path prefix",
+		PromptPrefix: ">",
+		AllowVars:    false,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       PathPrefix,
+		Callback:     nil,
+	},
+	SocketTimeoutMillis: {
+		Msg:          "Enter socket timeout millis",
+		PromptPrefix: ">",
+		AllowVars:    false,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       SocketTimeoutMillis,
+		Callback:     nil,
+	},
+}
+
+func jobTypeCallback(iq *utils.InteractiveQuestionnaire, jobType string) (string, error) {
+	iq.MandatoryQuestionsKeys = append(iq.MandatoryQuestionsKeys, CronExp)
+	if jobType == Pull {
+		iq.OptionalKeysSuggests = getAllPossibleOptionalRepoConfKeys()
+	} else {
+		iq.MandatoryQuestionsKeys = append(iq.MandatoryQuestionsKeys, Username, Password, URL)
+		iq.OptionalKeysSuggests = getAllPossibleOptionalRepoConfKeys()
+	}
+	return "", nil
+}
+
+func getAllPossibleOptionalRepoConfKeys(values ...string) []prompt.Suggest {
+	optionalKeys := []string{utils.WriteAndExist, Enabled, SyncDeletes, SyncProperties, SyncStatistics, PathPrefix, EnableEventReplication, SocketTimeoutMillis}
+	if len(values) > 0 {
+		optionalKeys = append(optionalKeys, values...)
+	}
+	return utils.GetSuggestsFromKeys(optionalKeys, suggestionMap)
+}
+
+//duplicate code
```

Comment:
What this comment mean?

Reply:
repo and replication use this code(duplicated) need to merge it carefully with  liron